File: Spring-2019/HW-5/p1/p1.go
package main

import (
	"fmt"
)

func main() {
	var n int

	fmt.Scanf("%d", &n)

	var index = 1
	res := digit(n, index)
	for res != -1 {
		fmt.Printf("%d ", res)
		index++
		res = digit(n, index)
	}
	fmt.Print("\n")
	for index > 1 {
		index--
		fmt.Printf("%d ", digit(n, index))
	}
	fmt.Print("\n")
}

func digit(n, m int) int {
	i := 0

	for n != 0 {
		i++
		if i == m {
			return n % 10
		}
		n /= 10
	}

	return -1
}

File: Fall-2020/Midterm/p1/p1.c
#include <stdio.h>

int check_sequence(int x) {
  int last_num = 1, n = 1;
  while (last_num < x) {
    n++;
    last_num += 2 * n;
  }
  if (last_num == x)
    return n;
  else
    return -1;
}

int main() {
  int num, res;
  scanf("%d", &num);
  res = check_sequence(num);
  printf("%d", res);
}
